components/ThemeComponents/ThemeContext.js

```diff
@@ -0,0 +1,50 @@
+import React, { useState } from 'react';
+import { getTheme } from './ThemeColorschemes';
+
+const ContextData = {
+  theme: '',
+  toggleTheme: () => {},
+};
+
+const ThemeContext = React.createContext(ContextData);
+const useTheme = () => React.useContext(ThemeContext);
+
+class ThemeToggleProvider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = '';
+  }
+
+  componentDidMount() {
+    this.state =
+      getComputedStyle(document.documentElement).getPropertyValue('--primary') == '#ffffff' ? 'light' : 'dark';
+
+  }
+
+  render() {
+    const switchTheme = mode => {
```

Comment:
Functions in class Components can be defined in the class body.

I'll leave this example here:

``` javascript
class Foo extends Component {
  state = { theme: 'light' };

  switchTheme = (mode) => {
    ...
  };

  render() {
    ...
  }
}
```

*Note: this exact syntax requires a babel plugin that might not be standard in NextJS, but if we don't have it we should probably install it*
https://www.npmjs.com/package/@babel/plugin-proposal-class-properties